templates/html_report/js/graph-maintainability.js

```diff
@@ -1,14 +1,23 @@
-function chartMaintainability(withoutComment)
-{
-    var chartId = withoutComment ? 'svg-maintainability-without-comments' : 'svg-maintainability';
-
+function chartMaintainability(withoutComment) {
+    var chartId = 'svg-maintainability';
+    withoutComment = typeof (withoutComment) !== 'undefined' ? withoutComment : false;
```

Comment:
I think you should either:
- force the default graph to be with comments and so, remove the ability to call the method `chartMaintainability` with an argument; or
- let the graph to be loaded being decided by the UI when the call is made, and keep the argument, but in this case, force the call to be with `true` or `false`.

Therefore, in any case, this default value setting for `withoutComment` should be removed.

Reply:
Hi,

sorry, I don't understand

> force the default graph to be with comments and so, remove the ability to call the method chartMaintainability with an argument; or
Now by default the graph is "with comments". But I need to pass an argument to change behavior on switch (via the `toggleChartMaintainability` function)

> let the graph to be loaded being decided by the UI when the call is made, and keep the argument, but in this case, force the call to be with true or false.

In this case it will be difficult to switch easily the kinf of chart by click.

I think I misunderstood your comment. Do you have any code suggestion to help me to understand ?

Reply:
Ok, I misread something in your code and I didn't noticed that `chartMaintainability` is called with an argument within `toggleChartMaintainability`.

So, my suggestion here is to call `chartMaintainability(false);` (give the argument) to force the `chartMaintainability` function to recieve an argument. It's clearer than defining a default value based on the fact the variable is defined or undefined:

```suggestion
```

And, both usages of `chartMaintainability();` in `templates/html_report/index.php` and `templates/html_report/panel.php` must be replaced by `chartMaintainability(false)`.

Reply:
Hi,

I suggest to merge this for the moment. I'll try to improve it after